packages/babel-plugin-jest-hoist/src/index.ts

```diff
@@ -152,38 +164,113 @@ FUNCTIONS.deepUnmock = args => args.length === 1 && args[0].isStringLiteral();
 FUNCTIONS.disableAutomock = FUNCTIONS.enableAutomock = args =>
   args.length === 0;
 
-export default (): {visitor: Visitor} => {
-  const shouldHoistExpression = (expr: NodePath): boolean => {
-    if (!expr.isCallExpression()) {
-      return false;
-    }
+const createJestObjectGetter = statement`
+function GETTER_NAME() {
+  const { JEST_GLOBALS_MODULE_JEST_EXPORT_NAME } = require("JEST_GLOBALS_MODULE_NAME");
+  GETTER_NAME = () => JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
+  return JEST_GLOBALS_MODULE_JEST_EXPORT_NAME;
+}
+`;
 
-    // TODO: avoid type casts - the types can be arrays (is it possible to ignore that without casting?)
-    const callee = expr.get('callee') as NodePath;
-    const expressionArguments = expr.get('arguments');
-    const object = callee.get('object') as NodePath;
-    const property = callee.get('property') as NodePath;
-    return (
-      property.isIdentifier() &&
-      FUNCTIONS[property.node.name] &&
-      (object.isIdentifier(JEST_GLOBAL) ||
-        (callee.isMemberExpression() && shouldHoistExpression(object))) &&
-      FUNCTIONS[property.node.name](
-        Array.isArray(expressionArguments)
-          ? expressionArguments
-          : [expressionArguments],
-      )
-    );
-  };
-
-  const visitor: Visitor = {
-    ExpressionStatement(path) {
-      if (shouldHoistExpression(path.get('expression') as NodePath)) {
-        // @ts-ignore: private, magical property
-        path.node._blockHoist = Infinity;
-      }
-    },
-  };
+const isIdentifierJestObject = (identifier: NodePath<Identifier>): boolean => {
+  // global
```

Comment:
not actually global, buy y'know 😀

Reply:
I call `window` global before someone `new Function(code, 'window')`s my `code` :P